engine/src/main/java/org/terasology/rendering/nui/layers/ingame/metrics/DebugMetricsSystem.java

```diff
@@ -95,8 +98,9 @@ public MetricsMode toggle() {
      * @return True if the MetricsMode instance was in the set. False otherwise.
      */
     public boolean unregister(MetricsMode mode) {
+        Preconditions.checkNotNull(mode, "Illegal argument passed, mode is null.");
         if (mode == defaultMode) {
-            throw new IllegalArgumentException("Removing defaultMode is not allowed!");
+            throw new IllegalArgumentException("Removing defaultMode is not allowed.");
```

Comment:
@Cervator: this is the exception I was asking you about, but I've come to the conclusion it is the right thing to do, so I won't ask confirmation after all.